tracer/src/Datadog.Trace/RemoteConfigurationManagement/RcmSubscriptionManager.cs

```diff
@@ -23,18 +27,18 @@ internal class RcmSubscriptionManager : IRcmSubscriptionManager
     public static readonly IRcmSubscriptionManager Instance = new RcmSubscriptionManager();
     private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<RcmSubscriptionManager>();
 
-    private readonly List<ISubscription> _subscriptions = new();
     private readonly object _syncRoot = new();
 
     /// <summary>
     /// Key is the path
     /// </summary>
-    private readonly Dictionary<string, RemoteConfigurationCache> _appliedConfigurations = new();
+    private readonly ConcurrentDictionary<string, RemoteConfigurationCache> _appliedConfigurations = new();
 
     private readonly string _id;
 
-    private string? _backendClientState;
-    private int _targetsVersion;
+    private IReadOnlyList<ISubscription> _subscriptions = new List<ISubscription>();
```

Comment:
```suggestion
    // Ideally this would be an ImmutableArray but that's not available in net461
    // so we just take care to _treat_ it as immutable
    private IReadOnlyList<ISubscription> _subscriptions = new List<ISubscription>();
```

Reply:
Added in https://github.com/DataDog/dd-trace-dotnet/pull/4996/commits/de6195058a6c95e9abc96b7febb56a266773db95